res/controllers/midi-controller-api.d.ts

```diff
@@ -29,4 +38,7 @@ declare namespace midi {
      */
     function sendSysexMsg(dataList: number[], length?: number): void;
 
+    type InputCallback = (channel: string, control: string, value: number, status: number, group: string) => void
```

Comment:
I think you can get rid of the group string. That only really makes sense for handlers defined in XML. but the entire reason for registering handlers from JS is that you don't need the XML (at least not the handler definitions).

Reply:
This is the signature of `components.Component.input`. I feel like it may be confusing for the user if this argument is undefined. Don't you think?

Reply:
Good point. I'm not really sure either. It doesn't really make sense for componentJS either and just came from signature of the XML-bound handlers.
I imagine usage of this API to look like this:

```js
midi.makeInputHandler(..., (channel, control, value, status) => {...} );
```

If this was bound to an XML-originating handler, you could just simply supply the group if you need it right then and there. 

```js
midi.makeInputHandler(..., (channel, control, value, status) => {
    someComponent.input(channel,control,value,status, someGroupFromEnclosingScope
} );
```

If thats too much to type, HOF got you covered:


```js
// once somewhere
const makeLegacyWithGroup = (group, handler) => 
    (channel,control,value,status) => handler(channel,control,value,status,group);

// then use it
midi.makeInputHandler(..., makeLegacyWithGroup(someGroupFromEnclosingScope, someComponent.input));
```

As I see it, the only reason the group was introduce in the first place was that there was more information available in the manually created XML than in the corresponding JS handler. Now that all the information is already available in JS (since thats where the handler is created), that loophole is not needed anymore. Do you agree?

Reply:
Sounds good to me. I'll just need to remember about documenting this behavior when this is merged.

Reply:
So if we don't distinguish anymore from anonymous and named function, I have to reintroduce this argument. At least for documentation consistency.

Reply:
yes... you're right, to some extend. I still don't think its super useful though. Why would the handler take a group, when there is no way to specify one from the handler creation function. I still don't think you should be able to specify one because its just unnecessary overhead in the C++ domain.

Reply:
lets keep the group for now, but please add some jsdoc comments and mark the group as deprecated, wdyt?